### PR TITLE
Add troubleshooting section to MPS docs

### LINKED ISSUE #14634: MPSAccelerator not available
## 🐛 Bug

This issue is similar to this one: https://github.com/Lightning-AI/lightning/issues/13991

> pytorch_lightning.utilities.exceptions.MisconfigurationException: MPSAccelerator can not run on your system since the accelerator is not available. The following accelerator(s) is available and can be passed into `accelerator` argument of `Trainer`: ['cpu'].

Raised from accelerator_connector.py::AcceleratorConnector#_set_parallel_devices_and_init_accelerator:
https://github.com/Lightning-AI/lightning/blob/2415834aaf4f215ddb0d618bf415405d670f9114/src/pytorch_lightning/trainer/connectors/accelerator_connector.py

Availability determined here:
https://github.com/Lightning-AI/lightning/blob/2415834aaf4f215ddb0d618bf415405d670f9114/src/pytorch_lightning/accelerators/mps.py

Check is wrong due to platform check here expecting arm (my M1 Pro reports platform i386):
https://github.com/Lightning-AI/lightning/blob/master/src/pytorch_lightning/accelerators/mps.py#L27

Screenshot

### To Reproduce

```
from pytorch_lightning import Trainer
Trainer(accelerator='mps')
```

### Expected behavior

Trainer should run without an error.

### Environment

```
* CUDA:
	- GPU:               None
	- available:         False
	- version:           None
* Lightning:
	- adabelief-pytorch: 0.2.1
	- efficientnet-pytorch: 0.7.1
	- pytorch-lightning: 1.7.5
	- pytorch-ranger:    0.1.1
	- pytorch-toolbelt:  0.5.2
	- torch:             1.12.1
	- torch-optimizer:   0.3.0
	- torchaudio:        0.12.1
	- torchmetrics:      0.8.2
	- torchvision:       0.13.1
* Packages:
	- absl-py:           1.0.0
	- adabelief-pytorch: 0.2.1
	- adabound:          0.0.5
	- aiohttp:           3.8.1
	- aiosignal:         1.2.0
	- albumentations:    1.2.1
	- antlr4-python3-runtime: 4.9.3
	- async-timeout:     4.0.2
	- attrs:             21.4.0
	- black:             22.3.0
	- cachetools:        5.1.0
	- certifi:           2022.5.18.1
	- cfgv:              3.3.1
	- charset-normalizer: 2.0.12
	- click:             8.1.3
	- colorama:          0.4.4
	- colorlog:          6.6.0
	- comet-ml:          3.31.12
	- commonmark:        0.9.1
	- configobj:         5.0.6
	- distlib:           0.3.4
	- docker-pycreds:    0.4.0
	- dulwich:           0.20.41
	- efficientnet-pytorch: 0.7.1
	- einops:            0.4.1
	- everett:           3.0.0
	- filelock:          3.7.0
	- flake8:            4.0.1
	- frozenlist:        1.3.0
	- fsspec:            2022.5.0
	- gitdb:             4.0.9
	- gitpython:         3.1.27
	- google-auth:       2.6.6
	- google-auth-oauthlib: 0.4.6
	- grpcio:            1.46.3
	- hydra-colorlog:    1.2.0
	- hydra-core:        1.2.0
	- identify:          2.5.1
	- idna:              3.3
	- imageio:           2.19.2
	- iniconfig:         1.1.1
	- joblib:            1.1.0
	- jsonschema:        4.5.1
	- markdown:          3.3.7
	- mccabe:            0.6.1
	- mkl-fft:           1.3.1
	- mkl-random:        1.2.2
	- mkl-service:       2.4.0
	- multidict:         6.0.2
	- munch:             2.5.0
	- mypy:              0.950
	- mypy-extensions:   0.4.3
	- networkx:          2.8.2
	- nodeenv:           1.6.0
	- numpy:             1.23.3
	- nvidia-ml-py3:     7.352.0
	- oauthlib:          3.2.0
	- omegaconf:         2.2.1
	- opencv-python:     4.5.5.64
	- opencv-python-headless: 4.5.5.64
	- packaging:         21.3
	- pandas:            1.4.4
	- pathspec:          0.9.0
	- pathtools:         0.1.2
	- pillow:            9.2.0
	- pip:               21.2.4
	- platformdirs:      2.5.2
	- pluggy:            1.0.0
	- pre-commit:        2.19.0
	- pretrainedmodels:  0.7.4
	- promise:           2.3
	- protobuf:          3.19.4
	- psutil:            5.9.1
	- py:                1.11.0
	- pyasn1:            0.4.8
	- pyasn1-modules:    0.2.8
	- pycodestyle:       2.8.0
	- pydeprecate:       0.3.2
	- pyflakes:          2.4.0
	- pygments:          2.12.0
	- pyparsing:         3.0.9
	- pyrsistent:        0.18.1
	- pytest:            7.1.3
	- python-dateutil:   2.8.2
	- pytorch-lightning: 1.7.5
	- pytorch-ranger:    0.1.1
	- pytorch-toolbelt:  0.5.2
	- pytz:              2022.1
	- pywavelets:        1.3.0
	- pyyaml:            6.0
	- qudida:            0.0.4
	- ranger-adabelief:  0.1.0
	- requests:          2.27.1
	- requests-oauthlib: 1.3.1
	- requests-toolbelt: 0.9.1
	- rich:              12.5.1
	- rsa:               4.8
	- scikit-image:      0.19.2
	- scikit-learn:      1.1.2
	- scipy:             1.8.1
	- semantic-version:  2.9.0
	- sentry-sdk:        1.5.12
	- setproctitle:      1.2.3
	- setuptools:        61.2.0
	- shortuuid:         1.0.9
	- six:               1.16.0
	- smmap:             5.0.0
	- tabulate:          0.8.9
	- tensorboard:       2.10.0
	- tensorboard-data-server: 0.6.1
	- tensorboard-plugin-wit: 1.8.1
	- threadpoolctl:     3.1.0
	- tifffile:          2022.5.4
	- timm:              0.6.7
	- toml:              0.10.2
	- tomli:             2.0.1
	- torch:             1.12.1
	- torch-optimizer:   0.3.0
	- torchaudio:        0.12.1
	- torchmetrics:      0.8.2
	- torchvision:       0.13.1
	- tqdm:              4.64.0
	- typing-extensions: 4.1.1
	- urllib3:           1.26.9
	- virtualenv:        20.14.1
	- wandb:             0.13.3
	- websocket-client:  1.3.2
	- werkzeug:          2.1.2
	- wheel:             0.37.1
	- wrapt:             1.14.1
	- wurlitzer:         3.0.2
	- yarl:              1.7.2
* System:
	- OS:                Darwin
	- architecture:
		- 64bit
		-
	- processor:         i386
	- python:            3.10.4
	- version:           Darwin Kernel Version 21.5.0: Tue Apr 26 21:08:37 PDT 2022; root:xnu-8020.121.3~4/RELEASE_ARM64_T6000
```

Additional details:
I checked, and if I add `i386` here: https://github.com/Birch-san/lightning/blob/13375b51ebda40d94fd37741627c4594776cccd1/src/pytorch_lightning/accelerators/mps.py#L28 the Trainer will work without any errors.

I could add a PR with this change, if necessary.

cc @akihironitta @justusschock

### PR DESCRIPTION
**Pull request by awaelchli**
## What does this PR do?

Fixes #14634

Adds a troubleshooting section. Added the exception too so Googlers can hopefully find our advice too. 

Note: We currently have no way to detect that MPS is available when running in an Intel emulated environment. Otherwise we would have a different error directly in Lightning. Suggestions regarding this welcome

## Before submitting
- [x] Was this discussed/approved via a GitHub issue? (not for typos and docs)
- [x] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), **Pull Request** section?
- [x] Did you make sure your PR does only one thing, instead of bundling different changes together?
- [x] Did you make sure to update the documentation with your changes? (if necessary)
- [x] Did you write any new necessary tests? (not for typos and docs)
- [x] Did you verify new and existing tests pass locally with your changes?
- [x] Did you update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/CHANGELOG.md)? (not for typos, docs, test updates, or internal minor changes/refactorings)

## PR review
Anyone in the community is free to review the PR once the tests have passed.
Before you start reviewing make sure you have read [Review guidelines](https://github.com/PyTorchLightning/pytorch-lightning/wiki/Review-guidelines). In short, see the following bullet-list:

 - [x] Is this pull request ready for review? (if not, please submit in draft mode)
 - [x] Check that all items from **Before submitting** are resolved
 - [x] Make sure the title is self-explanatory and the description concisely explains the PR
 - [x] Add labels and milestones (and optionally projects) to the PR so it can be classified

## Did you have fun?
I made sure I had fun coding 🙃

### CODE REVIEW

**Review by carmocca — APPROVED** [review]

**Review by akihironitta — APPROVED** [review]

File: docs/source-pytorch/accelerators/mps_basic.rst [hunk 1]
```diff
@@ -46,3 +46,18 @@ Enable the following Trainer arguments to run on Apple silicon gpus (MPS devices
...
+
+
+If Lightning can't detect the Apple Silicon hardware, it will raise this exception:
+
+.. code::
+
+   MisconfigurationException: MPSAccelerator can not run on your system since the accelerator is not available.
+
+If you are seeing this despite running on an ARM-enabled Mac, the most likely cause is that your Python is being emulated and thinks it is running on an Intel CPU.
+To solve this, re-install your conda by downloading the Apple M1/M2 build (not Intel!), for example `here <https://docs.conda.io/en/latest/miniconda.html#latest-miniconda-installer-links>`_.
```
Comment (justusschock) [inline docs/source-pytorch/accelerators/mps_basic.rst:63]:
```suggestion
To solve this, re-install your python executable (and if using environment managers like conda, you have to reinstall these as well) by downloading the Apple M1/M2 build (not Intel!), for example `here <https://docs.conda.io/en/latest/miniconda.html#latest-miniconda-installer-links>`_.
```

**Review by justusschock — APPROVED** [review]

**Review by Borda — APPROVED** [review]
